frontend/src/components/editor/actions/name-cell-input.tsx

```diff
@@ -0,0 +1,120 @@
+/* Copyright 2023 Marimo. All rights reserved. */
+import { Input } from "@/components/ui/input";
+import { getCellNames, useCellActions } from "@/core/cells/cells";
+import { CellId } from "@/core/cells/ids";
+import {
+  DEFAULT_CELL_NAME,
+  normalizeName,
+  getValidName,
+} from "@/core/cells/names";
+import { useOnMount } from "@/hooks/useLifecycle";
+import { cn } from "@/utils/cn";
+import { Events } from "@/utils/events";
+import React, { useRef, useState } from "react";
+
+interface Props {
+  className?: string;
+  value: string;
+  onChange: (newName: string) => void;
+  placeholder?: string;
+}
+
+export const NameCellInput: React.FC<Props> = ({
+  value,
+  onChange,
+  placeholder,
+}) => {
+  const ref = useRef<HTMLInputElement>(null);
+  const inputProps = useCellNameInput(value, onChange);
+
+  // Custom onBlur without React's synthetic events
+  // See https://github.com/facebook/react/issues/12363
+  useOnMount(() => {
+    const onBlur = inputProps.onBlur;
+    const input = ref.current;
+    if (!input) {
+      return;
+    }
+
+    input.addEventListener("blur", onBlur);
+    return () => {
+      input.removeEventListener("blur", onBlur);
+    };
+  });
+
+  return (
+    <Input
+      value={inputProps.value}
+      onChange={inputProps.onChange}
+      ref={ref}
+      placeholder={placeholder}
+      className="shadow-none! hover:shadow-none focus:shadow-none focus-visible:shadow-none"
+      onKeyDown={Events.onEnter(Events.stopPropagation())}
+    />
+  );
+};
+
+export const NameCellContentEditable: React.FC<{
+  cellId: CellId;
+  value: string;
+  className: string;
+}> = ({ value, cellId, className }) => {
+  const { updateCellName } = useCellActions();
+  const inputProps = useCellNameInput(value, (newName) =>
+    updateCellName({ cellId, name: newName })
+  );
+
+  return (
+    <span
+      className={cn("outline-1 outline-cyan-500/40", className)}
```

Comment:
Let's have the cyan outline show up on hover as well, to make it more obvious that it's editable